File: Samples/CslaLight/vb/RemotePortalWithMobileFactoryAndObjectFactory/Sample.Business.Server/Security/SamplePrincipal.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports Csla
Imports Csla.Security
Imports Csla.Core
Imports Csla.Serialization
Imports Csla.DataPortalClient
Imports System.ComponentModel
Imports System.Security.Principal

#If Not SILVERLIGHT = 1 Then
Imports System.Data.SqlClient
#End If

<Serializable()> _
Public Class SamplePrincipal
  Inherits BusinessPrincipalBase

  Private Sub New(ByVal identity As IIdentity)
    MyBase.New(identity)
  End Sub


#If Silverlight = 1 Then
  Public Sub New()
  End Sub
#Else
  Private Sub New()
  End Sub
#End If


#If Silverlight = 1 Then



  Public Shared Event SignalLogin As EventHandler(Of EventArgs)

  Public Shared Sub Login(ByVal userName As String, ByVal password As String, ByVal completed As EventHandler(Of EventArgs))
    AddHandler SamplePrincipal.SignalLogin, completed
    SampleIdentity.GetIdentity(userName, password, AddressOf HandleLogin)
  End Sub

  Private Shared Sub HandleLogin(ByVal sender As Object, ByVal e As DataPortalResult(Of SampleIdentity))
    If e.Object IsNot Nothing AndAlso e.Error Is Nothing Then
      SetPrincipal(e.Object)
    Else
      SetPrincipal(SampleIdentity.UnauthenticatedIdentity)
    End If
    RaiseEvent SignalLogin(e.Object, EventArgs.Empty)
  End Sub

#End If

  Private Shared Sub SetPrincipal(ByVal identity As Csla.Security.CslaIdentity)
    Dim principal As SamplePrincipal = New SamplePrincipal(identity)
    Csla.ApplicationContext.User = principal
  End Sub

  Public Shared Sub Logout()
    Dim identity As Csla.Security.CslaIdentity = SampleIdentity.UnauthenticatedIdentity()
    Dim principal As SamplePrincipal = New SamplePrincipal(identity)
    Csla.ApplicationContext.User = principal
  End Sub

  Public Overrides Function IsInRole(ByVal role As String) As Boolean
    Return CType(MyBase.Identity, ICheckRoles).IsInRole(role)
  End Function

End Class

File: Samples/CslaLight/cs/LocalDataPortalUsingWCFService/WcfService.Business.Client/Security/SamplePrincipal.cs
//INSTANT C# NOTE: Formerly VB.NET project-level imports:
using System;
using System.Linq;
using System.Collections.Generic;
using System.Net;
using System.Xml;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Documents;
using System.Windows.Ink;
using System.Windows.Input;
using System.Windows.Media;
using System.Windows.Media.Animation;
using System.Windows.Shapes;

using System.Text;
using Csla;
using Csla.Security;
using Csla.Core;
using Csla.Serialization;
using Csla.DataPortalClient;
using System.ComponentModel;
using System.Security.Principal;

#if !SILVERLIGHT
using System.Data.SqlClient;
#endif

namespace WcfService.Business.Client
{
  [Serializable()]
  public class SamplePrincipal : BusinessPrincipalBase
  {

    private SamplePrincipal(IIdentity identity)
      : base(identity)
    {
    }


#if SILVERLIGHT
	  public SamplePrincipal()
	  {
	  }
#else
    private SamplePrincipal()
    {
    }
#endif


#if SILVERLIGHT



	  public static event EventHandler<EventArgs> SignalLogin;

	  public static void Login(string userName, string password, EventHandler<EventArgs> completed)
	  {
		SamplePrincipal.SignalLogin += completed;
		SampleIdentity.GetIdentity(userName, password, HandleLogin);
	  }

	  private static void HandleLogin(object sender, DataPortalResult<SampleIdentity> e)
	  {
		if (e.Object != null && e.Error == null)
		{
		  SetPrincipal(e.Object);
		}
		else
		{
		  SetPrincipal(SampleIdentity.UnauthenticatedIdentity());
		}
		if (SignalLogin != null)
			SignalLogin(e.Object, EventArgs.Empty);
	  }

#endif

    private static void SetPrincipal(Csla.Security.CslaIdentity identity)
    {
      SamplePrincipal principal = new SamplePrincipal(identity);
      Csla.ApplicationContext.User = principal;
    }

    public static void Logout()
    {
      Csla.Security.CslaIdentity identity = SampleIdentity.UnauthenticatedIdentity();
      SamplePrincipal principal = new SamplePrincipal(identity);
      Csla.ApplicationContext.User = principal;
    }

    public override bool IsInRole(string role)
    {
      return ((ICheckRoles)base.Identity).IsInRole(role);
    }

  }

} //end of root namespace